arch/cortex-m33/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Shared implementations for ARM Cortex-M33 MCUs.
+
+#![crate_name = "cortexm33"]
+#![crate_type = "rlib"]
+#![no_std]
+
+use core::fmt::Write;
+
+pub mod mpu {
+    pub type MPU = cortexm::mpu::MPU<16, 32>; // Cortex-M7 MPU has 16 regions
```

Comment:
```suggestion
    pub type MPU = cortexm::mpu::MPU<16, 32>;
```

Reply:
As an aside: At some point we're going to have to do something different here, probably requiring a chip or board to instantiate this eventually. The m33 supports _up to_ 16 regions, not necessarily exactly 16:

<img width="771" alt="image" src="https://github.com/user-attachments/assets/1078f9ac-5e35-43c5-92dd-b7d244154160" />
<img width="663" alt="image" src="https://github.com/user-attachments/assets/2f7249df-a6e2-458d-b6d0-7a54da507c3d" />

That said, I believe this "up to" caveat technically applies to all cortex-m mpus, and we haven't run to hardware that does less than the max yet... it's a config option when instantiated the ARM IP, and probably not one that saves enough space for anyone really care about reducing from the default [for general purpose MCUs at least].